src/app/[entry_id]/page.jsx

```diff
@@ -0,0 +1,100 @@
+"use client"
+
+import React from 'react'
+import { useEffect, useState } from 'react'
+import { supabase } from '../lib/supabase'
+import Link from 'next/link'
+import Image from 'next/image'
+
+function Entry({ params }) {
+  const [entryData, setEntryData] = useState();
+  const entry_id = params.entry_id
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        let { data: diary_entry, error } = await supabase
+          .from('diary_entries')
+          .select()
+          .eq('id', entry_id)
+        if (error) {
+          throw error;
+        }
+        setEntryData(diary_entry[0])
+      } catch (error) {
+        console.error('Error fetching data from Supabase:', error.message);
+      }
+    };
+    fetchData();
+  }, []);
+
+  return (
+    <>
+      {entryData &&
```

Comment:
would probably simplify this and just do a `if (!entryData) return null` above this. Keeps things a tad cleaner :)

Reply:
Or `if (!entryData) <LoadingSpinnerOfSomeSort />`